fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedView.java

```diff
@@ -1371,31 +1415,37 @@ public boolean getPartitionNamesToRefreshForMv(Set<String> toRefreshPartitions,
         }
 
         // check mv's query rewrite consistency mode property only in query rewrite.
+        TableProperty.QueryRewriteConsistencyMode mvConsistencyRewriteMode =
+                TableProperty.QueryRewriteConsistencyMode.CHECKED;
         if (isQueryRewrite) {
-            TableProperty.QueryRewriteConsistencyMode mvConsistencyRewriteMode
-                    = tableProperty.getQueryRewriteConsistencyMode();
+            mvConsistencyRewriteMode = tableProperty.getQueryRewriteConsistencyMode();
             switch (mvConsistencyRewriteMode) {
                 case DISABLE:
                     return false;
-                case LOOSE:
+                case NOCHECK:
                     return true;
+                case LOOSE:
                 case CHECKED:
                 default:
                     break;
             }
         }
-
-        if (partitionInfo instanceof SinglePartitionInfo) {
-            return getNonPartitionedMVRefreshPartitions(toRefreshPartitions, isQueryRewrite);
-        } else if (partitionInfo instanceof ExpressionRangePartitionInfo) {
-            // partitions to refresh
-            // 1. dropped partitions
-            // 2. newly added partitions
-            // 3. partitions loaded with new data
-            return getPartitionedMVRefreshPartitions(toRefreshPartitions, isQueryRewrite);
+        if (mvConsistencyRewriteMode == TableProperty.QueryRewriteConsistencyMode.LOOSE) {
+            return getPartitionNamesToRefreshForMvInLooseMode(toRefreshPartitions);
```

Comment:
use `getPartitionNamesToRefreshForMvInLooseMode ` only when `isQueryRewrite ` is true since loose mode should not affect in refreshing mv.

Reply:
Yes, here only when isQueryRewrite is true, we can set the mvConsistencyRewriteMode to loose, otherwise it's checked